ykcapi/src/lib.rs

```diff
@@ -48,6 +52,116 @@ pub extern "C" fn __ykutil_get_llvmbc_section(res_addr: *mut *const c_void, res_
     }
 }
 
+struct Registers {
+    rax: usize,
+    rdx: usize,
+    rcx: usize,
+    rbx: usize,
+    rsi: usize,
+    rdi: usize,
+    rsp: usize,
+}
+
+impl Registers {
+    fn from_ptr(ptr: *const c_void) -> Registers {
+        Registers {
+            rax: Registers::read_from_stack(ptr, 0),
+            rdx: Registers::read_from_stack(ptr, 1),
+            rcx: Registers::read_from_stack(ptr, 2),
+            rbx: Registers::read_from_stack(ptr, 3),
+            rsi: Registers::read_from_stack(ptr, 4),
+            rdi: Registers::read_from_stack(ptr, 5),
+            rsp: Registers::read_from_stack(ptr, 6),
+        }
+    }
+
+    fn read_from_stack(rsp: *const c_void, off: isize) -> usize {
+        unsafe {
+            let ptr = rsp as *const usize;
+            ptr::read::<usize>(ptr.offset(off))
+        }
+    }
+
+    fn get(&self, id: u16) -> usize {
+        match id {
+            0 => self.rax,
+            1 => self.rdx,
+            2 => self.rcx,
+            3 => self.rbx,
+            4 => self.rsi,
+            5 => self.rdi,
+            6 => unreachable!("We currently have no way to access RBP of the previous stackframe."),
+            7 => self.rsp,
+            _ => unreachable!(),
+        }
+    }
+}
+
+/// Parses the stackmap and saved registers from the given address (i.e. the return address of the
+/// deoptimisation call).
+/// FIXME: Until we have the stopgap interpreter we simply print out the values we find.
+#[cfg(target_arch = "x86_64")]
+#[no_mangle]
+pub extern "C" fn yk_stopgap(
+    addr: *const c_void,
+    size: usize,
+    retaddr: usize,
+    rsp: *const c_void,
+) {
+    eprintln!("Stopgap");
```

Comment:
If this really is a jit-state print, I think it needs to be compile-time guarded like the others.

Reply:
Elaborate.

Reply:
The other jit-state prints are [guarded like this](https://github.com/ykjit/yk/blob/cef36509b8a31636b804e314ffa8c5732199ffeb/ykrt/src/mt.rs#L207-L208).

If this is a jit-state print too, it should be guarded in the same way.

This may not be a jit-state print of course. I'm not sure.

Reply:
Probably makes sense to guard it as it's only needed for the tests.

Reply:
Reworded in e42675f. We decided offline that adding the guards isn't worth it.

Reply:
OK.